mmv1/templates/terraform/custom_flatten/network_security_security_profile_url_filters_flatten.go.tmpl

```diff
@@ -0,0 +1,94 @@
+{{/*
+	The license inside this block applies to this file
+	Copyright 2024 Google Inc.
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/ -}}
+func flattenNetworkSecuritySecurityProfileUrlFilteringProfile(v interface{}, d *schema.ResourceData, config *transport_tpg.Config) interface{} {
+	if v == nil {
+		return nil
+	}
+	original := v.(map[string]interface{})
+	if len(original) == 0 {
+		return nil
+	}
+	transformed := make(map[string]interface{})
+	transformed["url_filters"] =
+		flattenNetworkSecuritySecurityProfileUrlFilteringProfileUrlFilters(original["urlFilters"], d, config)
+
+	// We check again the length after removing the default url_filter
+	if transformed["url_filters"].(*schema.Set).Len() == 0 {
+		return nil
+	}
+	return []interface{}{transformed}
+}
+func flattenNetworkSecuritySecurityProfileUrlFilteringProfileUrlFilters(v interface{}, d *schema.ResourceData, config *transport_tpg.Config) interface{} {
+	if v == nil {
+		return v
+	}
+
+	// We check if the user included the default filter in his config
+	resourceDataContainsDefaultFilter := false
+	resourceData, ok := d.GetOk("url_filtering_profile.0.url_filters")
+	if ok {
+		for _, raw := range resourceData.(*schema.Set).List() {
+			if raw.(map[string]interface{})["priority"] == 2147483647 {
+				resourceDataContainsDefaultFilter = true
+				break
+			}
+		}
+	}
+
+	l := v.([]interface{})
+	transformed := schema.NewSet(schema.HashResource(networksecuritySecurityProfileUrlFilteringProfileUrlFiltersSchema()), []interface{}{})
+	for _, raw := range l {
+		original := raw.(map[string]interface{})
+		if len(original) < 1 {
+			// Do not include empty json objects coming back from the api
+			continue
+		}
+
+		priorityFlatten := flattenNetworkSecuritySecurityProfileUrlFilteringProfileUrlFiltersPriority(original["priority"], d, config)
+		// Do not include the auto created default url_filter coming back from the api unless the user included it in his config
+		if priorityFlatten == 2147483647 && !resourceDataContainsDefaultFilter {
```

Comment:
I think this is causing the failure, the default priority is included in the config but we aren't seeing it in state after import:
=== CONT  TestAccNetworkSecuritySecurityProfile_networkSecuritySecurityProfileUrlFilteringUpdate
    resource_network_security_security_profile_test.go:121: Step 8/10 error running import: ImportStateVerify attributes not equivalent. Difference is shown below. The - symbol indicates attributes missing after import.
        
          map[string]string{
        - 	"url_filtering_profile.#":   "1",
        - 	"url_filtering_profile.0.%": "1",
          }

Reply:
Or the place above where we check for this priority specifically